contributors/design-proposals/api-proposal-design-template.md

```diff
@@ -0,0 +1,84 @@
+# Description of your Api
+
+## Owners
+
+Sig owner: list of sigs that will own the design, implementation, and support
+Individual owners: list of individual GitHub users that will own the design, implementation, and support
+
+## Abstract
+
+4-8 sentence description of the problem and solution.  Bullet points welcome.
+
+## Motivation
+
+4-8 sentence of the current state and why it is insufficient.
+
+## Use cases
+
+Bullet point list of use cases this proposal is meant to address.
+
+- As a cluster administrator...
+- As a application developer...
+- As a Kubernetes extension developer...
+
+## Dependencies and deadlines
+
+List of features that will depend on these changes.
+
+List of features that these changes will depend upon.
+
+**Include any deadlines that are related.** e.g. another feature depends on this and it is slated for 1.x.
+
+## Proposed changes for new types
+
+Declare the proposed Group / Version / Kind for any new types created as part of this proposal.
+
+Describe how the new types will be used by the end user.  Include sample yaml config and full walkthrough examples of
+1 or more usecases.
+
+### Defaulting
+
+Describe defaulting that will be performed on the new types
+
+### Validation
+
+Describe validation of fields on new types
+
+### Patching lists - strategic merge keys
+
+For any list fields, describe whether they will be replaced when patched or merged.  If merged,
+describe the merge key that will be used.
+
+### Controllers
+
+Describe any controllers that will be added or updated.
+
+## Proposed changes to existing types
+
+Declare any new fields for existing types and their version.  Will the fields be added as first class fields or
+as annotations?
+
+### Defaulting
+
+Describe any defaulting for new fields
+
+### Validation
+
+Describe any validation for new fields
+
+### Controllers
+
+Describe changes to controllers for new fields
+
+### Patching lists - strategic merge keys
+
+For any list fields, describe whether they will be replaced when patched or merged.  If merged,
+describe the merge key that will be used.
+
+## Impact on backwards / forwards compatibility
+
+Describe any impact on backwards / forwards compatibility
+
+## Alternatives considered
+
+List alternative solutions to the use cases.
```

Comment:
Especially client-side solutions and TPR.

And similarities to APIs in other similar systems.